### ISSUE TITLE
Any feature requests?

### ISSUE THREAD

**Issue by propyless** [propyless/panopuppet#70]
**Are there any features requests anyone here would like?**
I take all requests, small or big.

I know I have a few reported bugs/enhancements in the issues tracker which I have not got around to..

So if anyone has any feature requests applicable to PuppetDB 3.0 I am more than happy to take them on.

* Catalogue viewer.. (history and compare)
* Events viewer with a search function perhaps? (like query builder?) (I think I got this request a long time ago)
* Events Viewer right now shows all environments - Perhaps add the ability to view a specific environment?
* More graphs perhaps in the run time page perhaps?
  * Any specific data you would like to see graphs of? 
* Group based permissions with non-ldap authentication?

**Questions for my users:**
* Is anyone using the LDAP authentication?
* Is anyone using group based permissions that are available with LDAP Authentication?
* Are you solely using PanoPuppet or are you using it together with another dashboard?
 * If yes.. why? - Are there any features missing in PanoPuppet that exist in another Dashboard? 
* Any pages you feel are slow, ugly, or make you wonder why I did it like that?

**Comment by sheijmans** [comment]
**Questions for my users:**
`* Is anyone using the LDAP authentication?`
`* Is anyone using group based permissions that are available with LDAP Authentication?`
Not yet.

`* Are you solely using PanoPuppet or are you using it together with another dashboard?`
Not yet, using puppet-dashboard for reporting/ENC and puppetexplorer.

**Comment by sheijmans** [comment]
**Are there any features requests anyone here would like?**
Perhaps;
- Analytics --> Facts  
  able to see all available facts and select them, to see what contents is has over all nodes
- Option to disable Links, cannot access internet from the server
- Dashboard --> Recent  
  make this the homepage, instead of 'Welcome user'
- Node reports  
  able to select a date range, which to display
- Analytics --> Reports  
  Display all report and able to select a date range, which to display
- Display the total number of resources per node, in all the node overviews

**Comment by propyless** [comment]
Hello @sheijmans, thank you for your feedback.
Thank for the feature requests, a few I think I recieved before but had other priorities when creating the roadmap.

The welcome user page is a good point. It does not give much and is just "in the way" of the actual information thats of interest.
I can easily implement the node report date range function so that should be fairly easy to do. Same for the Analytics reports...
But from what I understand you are using the PuppetDB 2.x release of panopuppet? (at least that is my assumption based on the fact that you provided me with PRs to that branch.)
So I am also wondering, Normally I don't backport new features into that release. Its always not possible and also having to maintain two releases for just one of me is a bit too much work.

About the total number of resources per node. I am not entirely sure if that would be so optimal. But I do understand why that is something you want.
Since now with Puppetdb 3.x when querying each node specifically you recieve its resources and facts but since I don't do that and having to do that for each node in that selection. It would be a lot of overhead and I'm not sure if that would be worth it.

**Comment by sheijmans** [comment]
`>>But from what I understand you are using the PuppetDB 2.x release of panopuppet?`
Correct, but am preparing to upgrade to Puppet 4/PuppetDB 3.x soon, just wanted to have panopuppet up&running against the current version. So these request are for the current version of panopuppet.

`>>About the total number of resources per node. I am not entirely sure if that would be so optimal.`
`>> It would be a lot of overhead and I'm not sure if that would be worth it.`
I mostly use the number of resources to determine what has change lately (days/week) on a node.
And what about if it could be displayed for just 1 node, show the total number of resources from every run in a timegraph?

**Comment by propyless** [comment]
Yes, I just verified this and it is possible to see the total resource count for each report. Its also the same api endpoint I am using for each report on that page so that would be a more viable solution if you are up to compromising there. =)

**Comment by sheijmans** [comment]
No problem :)

**Comment by sheijmans** [comment]
Would it also be usefull to bring the puppetdb status under panopuppet?
http://docs.puppetlabs.com/puppetdb/3.2/images/perf-dash-large.png
http://docs.puppetlabs.com/puppetdb/3.2/maintain_and_tune.html

**Comment by propyless** [comment]
Hm, while it can seem relevant, I am reluctant to add support for that seeing as puppetlabs maintains a interface for that.

**Comment by sheijmans** [comment]
I know what you mean but I was thinking of  1 viewing point.

**Comment by propyless** [comment]
Ok. This is going to sound very 90's, but are you like looking for a iframe kind of thing? haha

**Comment by sheijmans** [comment]
No idea....just lazy to open multiple url's 😄

**Comment by propyless** [comment]
Analytics --> Reports
Display all report and able to select a date range, which to display
PR available for this #112  for Puppetdb 2.x branch. 0.x-dev has this update right now.
Will port it to the master branch as well...

**Comment by sheijmans** [comment]
@propyless 
Just did a checkout of the Puppetdb 2.x branch. 0.x-dev.
I don't see the Analytics --> Reports option but do see the Analytics --> Events Query Params Report from/to, looks and works nice.

**Comment by propyless** [comment]
Yeah, I didn't make one for the reports page, only for the analytics events query so what you are seeing is correct!

**Comment by propyless** [comment]
@sheijmans  - Some time has passed now and I'm wondering if you have any feedback for the analytics pages ( the filtering which I added a few months ago) Is there anything you think needs improvement?

**Comment by sheijmans** [comment]
@propyless, the only thing I tested on jan 27, was the date filter.
In which branch were the other changes made?

**Comment by propyless** [comment]
The changes are in master now, the date filter thing. But I was wondering if you feel if it needs any improvement at all.

**Comment by sheijmans** [comment]
No, I don't have any recommendations (yet) :)
